torch/_dynamo/variables/functions.py

```diff
@@ -30,16 +33,38 @@ def wrap_bound_arg(val, options):
         return variables.EnumVariable(val, **options)
     elif isinstance(val, (type, abc.ABCMeta)):
         return variables.UserDefinedClassVariable(val, **options)
+    elif istensor(val):
+        # Assumption: this is only called from InliningInstructionTranslator, in which case
+        # it is assured we will not 'reconstruct' this value in codegen, since the usage is
+        # entirely inlined.
+        # TODO - can we assert this? we expect to not reach this case for non-inline,
+        # since InstructionTranslator __init__ prepares VariableTrackers for args of top
+        # level function including defaults.
```

Comment:
Yes you can - check `_produce_guard_code` in guards has a prior example of assuring a caller.

Reply:
oh, crawling stack frames lol. yes.

otoh, i wrote that comment before i figured out how to set up the source/guard correctly.  now, i'm not sure if it is important to assert this or state the assumption.  i am thinking this codepath should work generally now?

Reply:
no longer an issue